### ISSUE TITLE
phpcs wants to rewrite a class constant named NULL

### ISSUE THREAD

**Issue by rimas-kudelis** [PHP-CS-Fixer/PHP-CS-Fixer#8057]
### Problem description

PHPCS erroneously attempts to replace a reference to class constant named `NULL` to `null.` This problem did not manifest in 3.57.0.

### Minimal reproducer

Good code:
```php
            SomeClass::someStaticMethod('foo', SomeOtherClass::NULL)->method($arg1, $arg2);
```

Suggested (wrong) fix:
```php
            SomeClass::someStaticMethod('foo', SomeOtherClass::null)->method($arg1, $arg2);
```

### Fixer version

3.58.0

### PHP version

8.1.28

### How do you run Fixer?

PHAR

### Contribution Checks

- [X] I have verified if this problem was already reported
- [X] I am familiar with ["Feature or bug?"](https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/blob/master/feature-or-bug.rst)

**Comment by kubawerlos** [comment]
PHPCS is here: https://github.com/squizlabs/PHP_CodeSniffer/tree/3.7.2/bin

If you, by mistake and/or ignorence, are referring to PHP CS Fixer, there is already fix for it: https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/pull/8055

**Comment by rimas-kudelis** [comment]
Yes, I was, by mistake, referring to PHP CS Fixer.

**Comment by kubawerlos** [comment]
BTW how did you:

> * [x]  I have verified if this problem was already reported

?

There are 2 lines added in https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/pull/8055/files and one of them is usage od class constant named `NULL`.

**Comment by rimas-kudelis** [comment]
I'll admit I just skimmed the first page of currently open issues. Tried searching for word "null", but the search yields 23 open issues at the moment, and searching for "null class constant" (without quotes) instead returns two seemingly irrelevant issues.
 
I didn't expect this bug would have been discovered already since not many people have constants named `NULL` (presumably), plus I assumed if this were a known and fixed problem, the fix would have been released already, since it's a regression of sorts?
